### PR TITLE
feat(TDQ-14786): support hybrid tcomp/database

### PR DESCRIPTION
**Pull request by qiongli**
display the icon in recyle bin
lock the node when open it
make tdqReportRun work
add duplicate

### CODE REVIEW

File: main/plugins/org.talend.cwm.management/src/main/java/org/talend/cwm/db/connection/ConnectionUtils.java [hunk 1]
```diff
@@ -1402,11 +1410,45 @@ public static String getHadoopClusterIDOfHive(IRepositoryViewObject viewObject)
...
             DatabaseConnection dbConn = (DatabaseConnection) conn;
-            jdbc = "JDBC".equals(dbConn.getDatabaseType());
+            return isTcompJdbc(dbConn.getDatabaseType());
         }
-        return jdbc;
+        return false;
+    }
+
+    /**
+     *
```
Comment (xqliu-talend) [inline main/plugins/org.talend.cwm.management/src/main/java/org/talend/cwm/db/connection/ConnectionUtils.java]: this is a public API, should add java doc here

File: main/plugins/org.talend.cwm.management/src/main/java/org/talend/cwm/db/connection/ConnectionUtils.java [hunk 2]
```diff
@@ -406,7 +408,8 @@ public static ReturnCode checkJdbcJarFilePathDriverClassName(DatabaseConnection
             }
             if (!driverJarNameList.isEmpty()) {
                 LinkedList<String> driverJarRealPaths = getDriverJarRealPaths(driverJarNameList);
-                if (driverJarRealPaths.isEmpty()) {
+                // only check the real path for Platform running
+                if (!Platform.isRunning() && driverJarRealPaths.isEmpty()) {
```
Comment (xqliu-talend) [inline main/plugins/org.talend.cwm.management/src/main/java/org/talend/cwm/db/connection/ConnectionUtils.java]: according to the comment, it should be "Platform.isRunning()" not "!Platform.isRunning()", right ?

File: main/plugins/org.talend.cwm.management/src/main/java/org/talend/cwm/db/connection/ConnectionUtils.java [hunk 3]
```diff
@@ -1402,11 +1410,45 @@ public static String getHadoopClusterIDOfHive(IRepositoryViewObject viewObject)
...
+            if (type.getType().equals(dbType)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     *
```
Comment (xqliu-talend) [inline main/plugins/org.talend.cwm.management/src/main/java/org/talend/cwm/db/connection/ConnectionUtils.java]: this is a public API, should add java doc here

File: main/plugins/org.talend.dataprofiler.core/src/org/talend/dataprofiler/core/ui/action/AbstractMetadataCreationAction.java [hunk 4]
```diff
@@ -36,7 +36,7 @@
  */
 public abstract class AbstractMetadataCreationAction extends Action implements ICheatSheetAction {
 
-    private static final int WIDTH = 550;
+    private static final int WIDTH = 780;
```
Comment (xqliu-talend) [inline main/plugins/org.talend.dataprofiler.core/src/org/talend/dataprofiler/core/ui/action/AbstractMetadataCreationAction.java]: why change the width here?

File: main/plugins/org.talend.cwm.management/src/main/java/org/talend/cwm/db/connection/ConnectionUtils.java (hunk 2 above)
Comment (qiongli) [reply main/plugins/org.talend.cwm.management/src/main/java/org/talend/cwm/db/connection/ConnectionUtils.java]: yes, you are right

File: main/plugins/org.talend.dataprofiler.core/src/org/talend/dataprofiler/core/ui/action/AbstractMetadataCreationAction.java (hunk 4 above)
Comment (qiongli) [reply main/plugins/org.talend.dataprofiler.core/src/org/talend/dataprofiler/core/ui/action/AbstractMetadataCreationAction.java]: please see TDQ-14784

**Pull request merged**